.github/workflows/on.pull-request.main.yaml

```diff
@@ -45,3 +45,37 @@ jobs:
           make build-controller SERVICE=$SERVICE
         env:
           SERVICE: ${{ matrix.service }}
+
+  test-controllers:
+    name: test controllers
+    strategy:
+      fail-fast: false
+      matrix:
+        service:
+         - s3
+         - ecr
+         - sns
+         - sqs
+         - elasticache
+         - dynamodb
```

Comment:
technically the dynamodb service controller has yet to be checked in, so this will be a NOOP until I have checked it in (this week)